gtk4/src/requested_size.rs

```diff
@@ -0,0 +1,28 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use glib::translate::*;
+
+#[derive(Debug, Copy, Clone)]
+#[repr(transparent)]
+pub struct RequestedSize(ffi::GtkRequestedSize);
```

Comment:
As you have to pass in one into `distribute_natural_allocation()`, this should probably get a constructor too and maybe setters? How else would you call that function?

Reply:
I added the setters, no idea how about a constructor for now as the RequestedSize contains a client pointer, and couldn't find what it should hold yet.

Reply:
Per mclassen, the pointer is to be used "however you see fits", I ended up passing a null pointer there for now. As alot of the implementations i found in gtk4's tree doesn't make use of it except for CellArea* stuff like https://github.com/GNOME/gtk/blob/4df7cc948c1a4ef9e3e6bde35f56b6f3b4128bd5/gtk/gtkcellareaboxcontext.c#L722 or https://github.com/GNOME/gtk/blob/949f22b8b2778c1f3e83b1e47d904fcad6a5b913/gtk/gtkcellareabox.c#L730-L743

I wonder if there's a way to pass "Any" here? or it doesn't make sense?

Reply:
Who is supposed to provide that pointer and who is supposed to use it, and who is supposed to call `distribute_natural_allocation()` on what kind of widget? :)

It seems like a way to pass arbitrary data around so we'll need to understand how it's supposed to be used or otherwise will run into crashes.

Reply:
I dropped the commit adding RequestedSize out of this PR, as I just noticed all the shaders stuff are not usable yet :-) I will get back to the RequestedSize later in the future by re-implementing a container as an example and figuring out how it should be used and whether it can be safe to bind or not.